cplusplus/VImage.cpp

```diff
@@ -60,11 +60,12 @@ VIPS_NAMESPACE_START
 std::vector<double>
 to_vectorv( int n, ... )
 {
-	std::vector<double> vector( n );
+	std::vector<double> vector;
+	vector.reserve( n );
 	va_list ap;
 
 	va_start( ap, n );
-	for( int i = 0; i < n; i++ )
+	for( std::vector<double>::size_type i = 0; i < size_t(n); i++ )
 		vector[i] = va_arg( ap, double );
```

Comment:
This is undefined behavior because `vector` starts as an empty vector and `vector.reserve( n );` only changes its `capacity()`, see: https://stackoverflow.com/a/13029377.

If you want to add elements with `push_back()` or `insert()`, use `reserve`.  If you want to access elements using `[]` or `at()`, use `resize` (or initialize the vector with the `vector(size_type N)` constructor).

Reply:
Good point, I switched it back to reserving space with the constructor.